lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -196,11 +201,34 @@ def generate(self, module):
                 self.remove()
                 module.fail_json(msg="%s" % to_native(e))
 
+        elif not self.isPublicKeyValid(module):
+            pubkey = module.run_command([module.get_bin_path('ssh-keygen', True), '-yf', self.path])
+            pubkey = pubkey[1].strip('\n')
+            try:
+                self.changed = True
+                with open(self.path + ".pub", "w") as pubkey_f:
+                    pubkey_f.write(pubkey + '\n')
+                os.chmod(self.path + ".pub", stat.S_IWUSR + stat.S_IRUSR + stat.S_IRGRP + stat.S_IROTH)
+            except IOError:
+                module.fail_json(
+                    msg='The public key is missing or does not match the private key. '
+                        'Unable to regenerate the public key.')
+            self.public_key = pubkey
+
+            if self.comment:
+                try:
+                    args = [module.get_bin_path('ssh-keygen',
+                                                True), '-q', '-c', '-C', self.comment, '-f', self.path]
+                    module.run_command(args, data='y')
+                except IOError:
```

Comment:
Now you need to update `self.public_key` another time.

Reply:
Why the self.public_key needs to be updated?
Only the comment is updated within the private and public keys.
The public key itself does not change.

Reply:
Ah sorry, I got confused and thought the pubkey also contains the comment. It doesn't, so everything's fine!

Reply:
Wait. You should not be confused. you are right. The public key contains the comment. And of course, the file is updated with the new comment once it changed.

What I meant with the question, is that in the module, the return output contains only the public key output without the comment.
The comment output will be returned separately as we agreed.